content/blog/2021-09-14-cats-and-dogs.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Switching the Dataset for Your Pre-trained Models
```

Comment:
btw, is it different (on a conceptual level) from the transfer learning blog post we put together before?

Reply:
> @dberenbaum @daavoo would you, guys, understand from this title the idea?

I'm not 100% sure if I get it. Are we pulling the weights of the "bees and ants" model and fine-tuning on a new "cats and dogs" dataset?

Reply:
Or is it more about we are going to take the same code used for the "bees and ants" model and reuse it for training with a new "dogs and cats" dataset?

Reply:
There has been a lot of focus on "data-centric" ML over "model-centric" ML lately, and this seems like it's addressing that (correct me if I'm wrong @flippedcoder). If so, I like the idea again and think it makes sense as a follow-up to the last one 🔥 !

However, I think it's unlikely that business problems change so dramatically that someone discards the original dataset and adopts a completely unrelated one. A more realistic scenario might be to analyze performance separately on bees and ants, and then add more bees if the model performs worse on those or vice versa (maybe you could start with a subset of the data if it's too hard to find additional bee/ant images 😄 ). A middle ground might be adding dogs and cats to the bees and ants to get a more generic animal classifier rather than discarding the original data completely.

Even better would be to combine these ideas and iterate on the data multiple times. For example, you could start with a subset of bees and ants, then add some cats, then some dogs, and finally some more of whatever animal is performing worst in the model.

I'm also not sure it's worth doing the hyperparameter tuning again in this post since it was already well covered last time and doesn't seem like it's the focus here.

I'll take a closer look once we narrow down the high-level flow of the post!

Reply:
> However, I think it's unlikely that business problems change so dramatically that someone discards the original dataset and adopts a completely unrelated one.

The way I understand this post is more of a transfer learning application. In this case it would make sense to take a pre-trained model and start fine-tuning it on the new dataset, right?

But then, what is the difference with the previous one?

Reply:
It's more about using DVC to handle data versioning, but I definitely see how that got lost in the way I wrote this. I really like @dberenbaum's idea of iterating on the data multiple time because it gets the point across better.

I'll work on this update and get it back to y'all in a day or two! Thanks for the really good feedback!

Reply:
If finding/extending the dataset seems to be difficult, you can update the post using `dataset-registry/fashion-mnist`. It has 10 classes and you can tell like "we have a socks and trousers classifier, now we add another class called shirts and see its performance."